beater/config.go

```diff
@@ -83,10 +87,12 @@ func (c *FrontendConfig) SmapMapper() (sourcemap.Mapper, error) {
 	if smap.mapper != nil {
 		return c.SourceMapping.mapper, nil
 	}
+	re := regexp.MustCompile("%.*{.*beat.version.?}")
```

Comment:
I think this method should be documented stating that it modifies its object, and that if a second time is called, nothing happens, ie. if i got it right this behaves like this:
```
c.beatVersion = 1
c.SmapMapper()
print  c.SourceMapping.index  // apm-1

c.beatVersion = 2
c.SmapMapper()
print  c.SourceMapping.index  // still apm-1
```

Reply:
I don't think that the `beat.version` can change (from a logical perspective) without restarting the server.

Reply:
But the point is the same. I don't say it's a bug, just that I would appreciate documentation when any kind of side effects are going on; or rename it to make it evident (eg `SetSourceMapperIfNil`)

Reply:
that's true, will change.

Reply:
@jalvz I did refactor a bit, please have a look.

As this PR doesn't introduce the problem, I suggest to create an Issue and address once we work on config reloading.

Good finding.

Reply:
LGTM. an issue is not necessary for me. It wasn't a comment on config reloading, but on code explicitness. thanks.